enhancements/storage/csi-migration.md

```diff
@@ -0,0 +1,145 @@
+   * Add the `CSIMigrationControlPlane` *FeatureSet* to the `featuregates/cluster` object:
+   ```shell
+   $ oc edit featuregates/cluster
+   (...)
+   spec:
+     featureSet: CSIMigrationControlPlane
+   ```
+   * Once all control-plane components have restart, add the `CSIMigrationNode` *FeatureSet* to the `featuresgates/cluster` object:
+   ```shell
+   $ oc edit featuregates/cluster
+   (...)
+   spec:
+     featureSet: CSIMigrationControlPlane, CSIMigrationNode
+   ```
+3. To disable CSI migration, the cluster administrator should perform the same steps in the opposite order:
+   * Add the `CSIMigrationNode` *FeatureSet* to the `featuregates/cluster` object.
+   * Wait for all Nodes to be drained and restarted.
+   * Add the `CSIMigrationControlPlane` *FeatureSet* to the `featuregates/cluster` object.
+4. It is the **responsability of the cluster administrator** to guarante the ordering described above is respected.
+
+### GA
+
+Once CSI migration reaches GA in upstream, the associated feature gates will be enabled by default and the features will not be optional anymore. As a result, CSI migration will be enabled by default in OCP as wel, and there will not be an option to disable it.
+
+In addition that, the *FeatureSets* created to handle the Tech Preview feature will no longer be operatinal. In that case, those *FeatureSets* should be ignored by operators.
+
+As for the required ordering described above, the [upgrade order](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/upgrades.md#generalized-ordering) performed by CVO during a cluster upgrade will take care of applying the feature gates in the correct order.
+
+#### Limitations
+
+The limitations of this approach lies on the downgrade process.
+
+In OCP, a downgrade is fundamentally a regular upgrade to an older version. That means that CVO downgrades components in the same order as it upgrades them: control-plane first, nodes later.
+
+However, as stated above, this ordering is **not** desired when **disabling** CSI migration, which is what is going to happen when the previous OCP version had CSI migration disabled.
+
+To address this issue we propose to document a simple workaround:
+
+1. Enable both `CSIMigrationNode`and `CSIMigrationControlplane` *FeatureSets*.
+1. Downgrade.
+
+As stated above, the *FeatureSets* `CSIMigrationNode`and `CSIMigrationControlplane` are not operational once CSI migration becomes GA. However, they will be carried over during the downgrade and they will be correctly applied once the system is downgraded.
+
+### Risks and Mitigations
+
+Although this approach does what we need, it has some drawbacks:
+
+1. Having operators ignoring certain *FeatureSets* is not common and error-prone.
+1. We need to patch many operators in order to ignore *FeatureSets*.
+
+## Design Details
+
+### Test Plan
+
+#### E2E jobs
+
+We want to create E2E jobs for all migrated plugins.
+
+For each E2E job:
```

Comment:
IMO, we want a separate job that runs a StatefulSet, flips the migration feature gates in the right order & waiting in between and checks that the StatefulSet survives. At least as stretch goal for tech preview.

Reply:
Added